Sources/Sentry/SentryHub.m

```diff
@@ -200,20 +205,29 @@ - (SentryId *)captureMessage:(NSString *)message withScope:(SentryScope *_Nullab
 
 - (SentryId *)captureError:(NSError *)error withScope:(SentryScope *_Nullable)scope
 {
-    [self incrementSessionErrors];
+    SentrySession *currentSession = [self incrementSessionErrors];
     SentryClient *client = [self getClient];
     if (nil != client) {
-        return [client captureError:error withScope:scope];
+        if (nil != currentSession) {
+            return [client captureError:error withSession:currentSession withScope:scope];
+        } else {
+            return [client captureError:error withScope:scope];
+        }
     }
     return SentryId.empty;
 }
 
 - (SentryId *)captureException:(NSException *)exception withScope:(SentryScope *_Nullable)scope
 {
-    [self incrementSessionErrors];
+    SentrySession *currentSession = [self incrementSessionErrors];
```

Comment:
L: Maybe we can throw the `getClient` on the top of the method and just return if there's none?

Reply:
But even if there is no client I think we still should increment the errors of the session.